units/spawnhost_termination_alert.go

```diff
@@ -0,0 +1,112 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/notify"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+)
+
+const (
+	spawnhostTerminationAlertName = "spawnhost-termination-alert"
+
+	emailSubject = "Spawn Host Terminated"
+	emailBody    = "Your spawn host '%s' has been terminated by Evergreen because it failed to provision. " +
+		"Feel free to spawn another."
+)
+
+func init() {
+	registry.AddJobType(spawnhostTerminationAlertName, func() amboy.Job {
+		return makeHostMonitorExternalState()
+	})
+}
+
+type spawnhostTerminationAlertJob struct {
+	job.Base `bson:"base" json:"base" yaml:"base"`
+
+	User string `bson:"user"`
+	Host string `bson:"host"`
+
+	// for testing
+	UseMock   bool              `bson:"use_mock,omitempty"`
+	sentMails []notify.MockMail `bson:"-"`
```

Comment:
I would get rid of this, and have one member `mailer notify.Mailer`. In the body of Run, if this is nil, set it with the real email sender from the notify package. 

In your tests, set the mailer to the mock mailer, and introspect that.

Reply:
done (but with grip)